eth/filters/statediff.go

```diff
@@ -0,0 +1,114 @@
+package filters
+
+import (
+	"math/big"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// ProcessStateChanges builds the state diff Payload from the modified accounts in the StateChangeEvent
+var emptyPayload Payload
+
+func ProcessStateChanges(event core.StateChangeEvent, crit ethereum.FilterQuery) (Payload, error) {
+	var accountDiffs []AccountDiff
+	block := event.Block
+	// Iterate over state changes to build AccountDiffs
+	for addr, modifiedAccount := range event.StateChanges {
+		if len(crit.Addresses) > 0 && !includes(crit.Addresses, addr){
```

Comment:
`ethereum.FilterQuery` also includes fields for `BlockHash`, `FromBlock`, `ToBlock` and `Topics` - thinking that we could potentially include the ability to filter by these as well. Not sure if it's possible to filter by `Topics`, but that would pretty cool if we could! Will probably add a TODO `BlockHash` and the block numbers for now, but could also be talked into handling this before opening an upstream PR.